debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java

```diff
@@ -157,26 +156,16 @@ protected void doStart() {
 
         // The 'source' object holds the starting point in the binlog where we should start reading,
         // set set the client to start from that point ...
-        String gtidSetStr = source.gtidSet();
-        if (gtidSetStr != null) {
+        GtidSet filteredGtidSet = context.getFilteredGtidSet();
+        if (filteredGtidSet != null) {
             // Register the event handler ...
             eventHandlers.put(EventType.GTID, this::handleGtidEvent);
-
-            logger.info("GTID set from previous recorded offset: {}", gtidSetStr);
-            // Remove any of the GTID sources that are not required/acceptable ...
-            Predicate<String> gtidSourceFilter = context.gtidSourceFilter();
-            if (gtidSourceFilter != null) {
-                GtidSet gtidSet = new GtidSet(gtidSetStr).retainAll(gtidSourceFilter);
-                gtidSetStr = gtidSet.toString();
-                logger.info("GTID set after applying GTID source includes/excludes: {}", gtidSetStr);
-                source.setGtidSet(gtidSetStr);
-            }
-            client.setGtidSet(gtidSetStr);
-            gtidSet = new com.github.shyiko.mysql.binlog.GtidSet(gtidSetStr);
+            logger.info("Registering binlog reader with GTID set: {}", filteredGtidSet);
+            client.setGtidSet(filteredGtidSet.toString());
+            gtidSet = new com.github.shyiko.mysql.binlog.GtidSet(filteredGtidSet.toString());
```

Comment:
This no longer calls `source.setGtidSet(...)`, so I think the best place to do that is here.